lib/git_worktree.rb

```diff
@@ -215,14 +217,14 @@ def checkout(target_directory)
     @repo.checkout_tree(tree, :target_directory => target_directory, :strategy => :force)
   end
 
-  def with_credential_options
+  def with_remote_options
     if @ssh_private_key
       @ssh_private_key_file = Tempfile.new
       @ssh_private_key_file.write(@ssh_private_key)
       @ssh_private_key_file.close
     end
 
-    options = {:credentials => method(:credentials_cb)}
+    options = {:credentials => method(:credentials_cb), :proxy_url => @proxy_url}
```

Comment:
I didn't double check this in the `rugged` changes, but when `nil`, this treats it as if it isn't there, correct?

Just want to make sure that `rugged` is looking for an existence of a `value`, and not the presence of a `key`.

Reply:
Went and looked it up:

https://github.com/ManageIQ/rugged/blob/4b3d493a/ext/rugged/rugged_repo.c#L416
https://github.com/ManageIQ/rugged/blob/4b3d493a/ext/rugged/rugged_remote.c#L220

And the answer is "yes, we are checking for the value":